src/legends/swatches.js

```diff
@@ -12,14 +12,20 @@ function maybeScale(scale, key) {
   return s;
 }
 
-export function legendSwatches(color, options) {
+export function legendSwatches(color, {opacity, ...options}) {
   if (!isOrdinalScale(color) && !isThresholdScale(color))
     throw new Error(`swatches legend requires ordinal or threshold color scale (not ${color.type})`);
   return legendItems(
     color,
     options,
     (selection, scale) =>
-      selection.append("svg").attr("fill", scale.scale).append("rect").attr("width", "100%").attr("height", "100%"),
+      selection
+        .append("svg")
+        .attr("fill", scale.scale)
+        .append("rect")
+        .attr("width", "100%")
+        .attr("height", "100%")
+        .attr("opacity", maybeNumberChannel(opacity)[1]),
```

Comment:
We can implement the option as fill-opacity here (for faster rendering, not that it is likely to matter in this case).

```suggestion
        .attr("fill-opacity", maybeNumberChannel(opacity)[1]),
```

Reply:
yup, why not; but then maybe also document that (by adding "fill" in README and in the jsdocs?)

Reply:
I’m not sure we need to document it because it’s an implementation detail how the swatches are rendered… but sure.